src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,127 @@
+
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), false);
+        }
+
+        //public static Tensor<T> Create(T* address, ReadOnlySpan<nint> lengths);
+        //public static Tensor<T> Create(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides);
+
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), mustPin);
+        }
+
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), mustPin);
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths) => new SpanND<T>(address, lengths, false);
+
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new SpanND<T>(address, lengths, false, strides);
+
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths) => new ReadOnlySpanND<T>(address, lengths, false);
+
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new ReadOnlySpanND<T>(address, lengths, false, strides);
+
+        public static Tensor<T> FillRange<T>(IEnumerable<T> data)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            T[] values = data.ToArray();
+            return new Tensor<T>(values, [values.Length], false);
+        }
+
+        // REVIEW: BASICALLY SEEDING A TENSOR. CHANGE API SHAPE/NAME?
+        public static Tensor<T> Uniform<T>(params nint[] lengths)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>, IFloatingPoint<T>
+        {
+            var linearLength = SpanHelpers.CalculateTotalLength(ref lengths);
+            T[] values = new T[linearLength];
+            Random rand = new Random();
+            for (int i = 0; i < values.Length; i++)
+                values[i] = T.CreateChecked(rand.NextSingle());
+
+            return new Tensor<T>(values, lengths, false);
+        }
+
+        #region Normal
+        // REVIEW: BASICALLY SEEDING A TENSOR. CHANGE API SHAPE/NAME?
+        public static Tensor<T> Normal<T>(params nint[] lengths)
```

Comment:
Dittos

Reply:
Renamed to all start with Create. Would appreciate your thoughts on the new names.